### ISSUE TITLE
BundlerMinifier does not work after update Visual Studio

### ISSUE THREAD

**Issue by wimbaens** [madskristensen/BundlerMinifier#536]
### Installed product versions
- Visual Studio: 2019 16.8.5
- This extension: 3.2.499

No minification happening since the latest Visual Studio Update.
Menu item is gone on right click solution. Also no bundleminifier options on right click bundlefile.
![2021-02-16 09_52_46-Window](https://user-images.githubusercontent.com/4004108/108039733-c9beb780-703c-11eb-85b4-e6135685e630.png)

**Comment by jweschenfelder** [comment]
Hello, we have experienced exaclty the same issue.

**Comment by chrisofthecoffee** [comment]
Hi I'd like to report the same behaviour here, also with BundlerMinifier.Core 3.2.499 on Visual Studio 2019 16.8.6

Not sure when it broke but I recently had to completely reinstall Windows and re-get my entire source from git, so I kinda expect that's when it started not working - I only notice the problem now a month later as I had to change some javascript, first js change since the Windows rebuild.

Fairly complex bundleconfig.json, it used to happily generate four different 'whatever-bundled.min.js' from various combinations of several 'whatever.js' files when I built but no longer doing this.

No context menu appears on the bundleconfig.json file, I tried deleting one of the generated the whatever-bundled.min.js files and nothing I do regenerates it.

I've tried altering the bundleconfig.json slightly to give it a prod, rebuilt in dev, rebuild in release configuration, restart VS, reinstall bundlerminifier, swear and curse, and repeat same steps again, same behaviour.

I read a solution about the javascript being broken, but it runs fine, the files that make up the bundle are run from the original js in development so that shouldn't be causing a problem with the bundling (and wouldn't stop the context menu item appearing either I guess). 

Cheers, Chris

**Comment by chrisofthecoffee** [comment]
Hi, absolutely no pressure, I appreciate the work that's gone into this fantastic free product and the support and all that.... but... is this being looked at soon ? If not then no problem, I'll switch to another addon or do some other workaround but I'd prefer to avoid rejigging a bunch of my projects so i can release new versions, then find you fixed this the next day :)

**Comment by jweschenfelder** [comment]
Same for us. If there is no update soon, we consider to switch to another bundler & minifier. For now we are doing it via manual process with https://www.cssportal.com/scss-to-css/ and https://cssminifier.com/ which cannot be the solution.

**Comment by jtsom** [comment]
This has been an issue for over 2 years now - https://github.com/madskristensen/BundlerMinifier/issues/391 and it's affecting our Azure DevOps build pipeline. We do not have the extension loaded, just the NuGet packages. Our only option is to not minify our javascript files.

**Comment by chrisofthecoffee** [comment]
>Our only option is to not minify our javascript files.
That will be my emergency solution if I have to make a new release with changed javascript, not entirely optimal. 
Damn, if it's a long-time issue I guess I'll have to start looking around at alternatives, ta for the info.

**Comment by jweschenfelder** [comment]
I had already worked with Laravel Mix in my former company to bundle and minify JS and SCSS/CSS. The problem is that you need the Node.js then which is not small (Node.js is practically already included in a VS 2019, but needs to be explicitly selected when installing VS 2019). Laravel Mix is similar to Gulp. BundlerMinifier is far easier to handle than Laravel or Gulp and the configuration files for BundlerMinifier look less complex, too. I would really prefer if someone would fix this issue.

**Comment by chrisofthecoffee** [comment]
I've used this and WebCompiler by the same Mads chap for some time and they've been fantastic, really solid and simple to implement even the most non-trivial setups. I tend to have different bundles for different types of user, no need for a guest to get all the javascript for the admin/back office areas that they'll never execute, for obvious size and security reasons.it'd be a big shame to switch look to something more complex or not as smooth and functional.

**Comment by jweschenfelder** [comment]
Meanwhile I would guess that it might have to do with Microsoft's switch to .NET Core and .NET Standard why this NuGet package is not updated yet. Ironically, the following package for .NET Core with Dependency Injection seems to be more up-to-date: https://github.com/ligershark/WebOptimizer.Sass (updated a few days ago).

**Comment by chrisofthecoffee** [comment]
I just needed to change the text of a javascript generated field validation message on a dialogue.... so edited the source javascript and also edited the generated bundled-xxxxx.min.js file... so I got by this time but anything much more involved and I'll need to find another bundler, damn.

Please can we have an indication of whether this problem is going to be looked at or not, no drama if not, that's absolutely fine, you provide this very useful addon for free and it's much appreciated; but the silence is making it hard to decide what the best way forward is for myself and others who have built systems around this great bundler.

**Comment by chrisofthecoffee** [comment]
I found a partial-but-good-enough solution.

I was missing / needed to install the BuildBundlerMinifier package in addition to BundlerMinifier.Core. 

I can't find any documentation that says you need both, maybe that should be obvious to me, but it wasn't.

Not a complete fix, the context menu still doesn't appear on the bundleconfig.json file but the bundles in it are regenerated at build time, which is good enough for me. Hooray.

**Comment by jweschenfelder** [comment]
Please keep the issue open because I could not get it running. And we had BuildBundlerMinifier installed as a NuGet package. Unless it is solved for wimbaens now, too. Then it might be a local installation problem. Since I have left the company and the project, the issue has solved for me in another way.

**Comment by chrisofthecoffee** [comment]
> Please keep the issue open because I could not get it running. And we had BuildBundlerMinifier installed as a NuGet package. 
> Unless it is solved for wimbaens now, too.

Yep, not my decision to close it, just saying it's resolved for me as far as it needs to be; I hope it's the solution for wimbaens too.

**Comment by jweschenfelder** [comment]
Update: I see right now, that a PR was made for .NET 5.0 Support today. We were already using .NET 5.0. Maybe that was the reason: https://github.com/madskristensen/BundlerMinifier/pull/550#pullrequestreview-627853334

**Comment by jtsom** [comment]
Unfortunately, looking at the code associated with the PR, it has nothing to do with fixing the issue - just adds .NET 5 support.

**Comment by wimbaens** [comment]
The installation of the BuildBundlerMinifier package did fix the minification on build. 
But the menu is still missing, I keep the issue open unless someone finds a solution for the second part.

**Comment by chrisofthecoffee** [comment]
It's stopped working for me again on build, I'd lost the context menu way back but was still OK as it was re-bundling on build until I'm not sure when. I tried reinstalling the Core and Build... packages but it's still not working.

The javascript is all being processed fine as per bundleconfig.json but compilerconfig.json isn't being acted upon to produce css files.

I'm about to handcraft a layout fix into the .css so I can make a release, then investigate this further but this is becoming a liability, the project it's used in is essentially complete so I only discover it's stopped working again when I need to make a layout fix.

There are a few new issues raised in the last month but I'm not sure anyone is supporting this any more, I ought to look at alternatives, shame, I really like BundlerMinifier.
 
Usual caveats about not wanting to gripe too much about open source code that's been fantastic until 6 months ago and that I've not put any work into other than raising issues, but it'd be good to know if it's going to be developed any more or if grateful users need to reluctantly say thanks for all the bundling and look for alternatives.

**Comment by chrisofthecoffee** [comment]
I got it working again ! - am sorry to say I don't have a '5 steps to make the world right again' procedure, I removed and reinstalled the three packages a few times over the last couple of days, deleted (well, moved) and reinstated bundleconfig.json and compilerconfig.json and swore at it, not sure which of these did the trick, I do remember that I installed BuildBundlerMinifier first, then BundlerMinifier.Core, then WebCompiler last.

My setup now works as before, right now at least; i.e. I don't get a context menu item off the main .scss files but it is properly processing the long list of scss imports in my 4 top level scss files and minifying on build, which is more than good enough for me.

My package references are as below, using Visual Studio 16.10.3;
![image](https://user-images.githubusercontent.com/17450109/124699824-2a805b00-df1e-11eb-86db-4c4373b03b8f.png)

I now get the scss files being processed on build;
![image](https://user-images.githubusercontent.com/17450109/124700584-8dbebd00-df1f-11eb-942f-9a4fc90dd87c.png)

Funnily enough today I had decided to bite the bullet and install the VS2019 extensions feature to try rebuilding the extensions locally to see if I could at least add some scratch tracing to see what the problem was. I didn't have that installed last time it worked so probably makes no difference but you never know. There is always the possibility that the code sensed with a deep, cold dread that I was about to get my gnarled fingers around its neck and decided to capitulate !

**Comment by Jackson-Calixto** [comment]
I'm not sure if the problem was that or not, in my case, I already had added the budleconfig.json before adding the Bundler & Minifer extension.
Then installed the extension and closed, reopened many times, Then installed BuildBundlerMinifier first, then BundlerMinifier.Core and restarted the VS again and nothing happened. 
Finally I deleted the budleconfig.json and added again then suddenly the X show up and now the context menu is there...

**Comment by TehWardy** [comment]
From what we could tell the problem is something like ...
Clean the solution -> generated files get removed
Build locally -> they get added again
Commit -> files missing on release

Clean the solution -> generated files get removed
Commit -> files available on released deployed app

Not sure why ... maybe some weirdness / difference between local and build environment.
I presume that's what others are seeing though.

**Comment by hn1** [comment]
We are also having serious issues with this package. Basically it does not work. Can someone suggest an alternative as this extension is not useful anymore.
Why Microsoft does not provide a PROPER built-in solution for such an essential task in Visual studio in this day and age is just beyond belief!
